### PR TITLE
New simulator communication hub.

### PR DESCRIPTION
**Pull request by rolfvdhulst**
This adds a new hub for the new simulator (converting robotcommands into the format it accepts). Please check if I did not break any functionality. That makes grSim compatible.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -27,7 +27,7 @@ add_executable(roboteam_robothub
         src/GRSim.cpp
         src/SerialDeviceManager.cpp
         src/RobotHub.cpp
-        src/main.cpp)
+        src/main.cpp src/MimirCommander.cpp include/roboteam_robothub/MimirCommander.h)
```
Comment (mrlukasbos) [inline CMakeLists.txt:30]: no need to include .h file

File: include/roboteam_robothub/MimirCommander.h [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    public:
+        /*
+         * @param sets the color of our team so we send to the correct team
+         */
+        void setColor(bool isYellow);
+        void setIP(const std::string &mimirIP);
+        void setPort(quint16 port);
+        bool sendCommand(const roboteam_proto::RobotCommand& robotCommand);
+    private:
+        void convertCommand(const roboteam_proto::RobotCommand& robotCommand,roboteam_proto::mimir_robotcommand &mimirCommand);
```
Comment (mrlukasbos) [inline include/roboteam_robothub/MimirCommander.h]: Why the pass by reference? Why not create the object inside this function?

File: src/MimirCommander.cpp [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+            sendPort
+    );
+    return sentBytes==dgram.size();
+}
+
+void MimirCommander::convertCommand(const roboteam_proto::RobotCommand &robotCommand, roboteam_proto::mimir_robotcommand &mimirCommand) {
+    mimirCommand.set_id(robotCommand.id());
+    mimirCommand.set_teamisyellow(weAreYellow);
+    //TODO: fix kicker commands
+    mimirCommand.mutable_kicker()->set_chip(true);
```
Comment (mrlukasbos) [inline src/MimirCommander.cpp:38]: this is shady

File: src/MimirCommander.cpp [hunk 4]
```diff
@@ -0,0 +1,48 @@
+//
+// Created by rolf on 27-10-19.
+//
+
+#include "MimirCommander.h"
+#include "iostream"
+namespace rtt{
+namespace robothub{
+void MimirCommander::setColor(bool isYellow) {
```
Comment (mrlukasbos) [inline src/MimirCommander.cpp:9]: make sure you format your code correctly. spaces/newlines are different everywhere

File: src/RobotHub.cpp [hunk 5]
```diff
@@ -76,7 +78,8 @@ void RobotHub::processRobotCommand(roboteam_proto::RobotCommand & cmd) {
     if (mode == utils::Mode::SERIAL) {
         sendSerialCommand(llrc);
     } else {
-        sendGrSimCommand(cmd);
+        //sendGrSimCommand(cmd);
```
Comment (mrlukasbos) [inline src/RobotHub.cpp:87]: Maybe just as well remove GrSim support at all? Or maybe make something nice to switch between versions. Or send to both?

Comment (rolfvdhulst) [reply src/RobotHub.cpp:87]: I think just adding an additional mode to settings would be best, actually.

**Comment by Yuhanun** [comment]
old, archiving

**Pull request closed without merging**